pulpcore/cli/rpm/prune.py

```diff
@@ -0,0 +1,89 @@
+import gettext
+import typing as t
+
+import click
+from pulp_glue.common.context import PulpException
+from pulp_glue.rpm.context import PulpRpmPruneContext, PulpRpmRepositoryContext
+
+from pulpcore.cli.common.generic import (
+    PulpCLIContext,
+    pass_pulp_context,
+    pulp_command,
+    resource_option,
+)
+
+_ = gettext.gettext
+
+multi_repository_option = resource_option(
+    "--repository",
+    "repositories",
+    default_plugin="rpm",
+    default_type="rpm",
+    context_table={"rpm:rpm": PulpRpmRepositoryContext},
+    multiple=True,
+    href_pattern=PulpRpmRepositoryContext.HREF_PATTERN,
+    help=_(
+        "RPM Repository to prune, in the form 'rpm:rpm:<name>' or by href."
+        " Can be called multiple times."
+    ),
+)
+
+
+@pulp_command()
+@multi_repository_option
+@click.option(
+    "--all-repositories",
+    type=bool,
+    is_flag=True,
+    show_default=True,
+    default=False,
+    help=_("Prune *all* repositories accessible to the invoking user."),
+)
+@click.option(
+    "--keep-days",
+    type=int,
+    default=14,
+    help=_("Prune packages that were added to the specified repositories more than N days ago."),
+)
+@click.option(
+    "--dry-run",
+    type=bool,
+    is_flag=True,
+    show_default=True,
+    default=False,
+    help=_("Evaluate the prune-status of the specified repositories but DO NOT make any changes."),
+)
+@pass_pulp_context
+def prune_packages(
+    pulp_ctx: PulpCLIContext,
+    repositories: t.Iterable[str],
```

Comment:
I think, this should be:
```suggestion
    repositories: t.Iterable[PulpRpmRepositoryContext],
```
And I'm sad, click's annotation does not catch these.